test/integration/collection-management/collection.test.ts

```diff
@@ -26,90 +26,38 @@ describe('Collection', function () {
+      expect(collection.collectionName).to.equal('test_collection_methods');
+      // Let's check that the collection was created correctly
+      const documents = await db.listCollections().toArray();
+      expect(documents).to.be.an('array');
+      expect(documents).to.have.length(1);
+      const nameArray = documents.map(doc => doc.name);
+      expect(nameArray).to.include('test_collection_methods');
+      // Rename the collection and check that it's gone
+      const renameCollection = await db.renameCollection(
+        'test_collection_methods',
+        'test_collection_methods2'
+      );
+      expect(renameCollection.collectionName).to.equal('test_collection_methods2');
+      // Drop the collection and check that it's gone
+      const drop = await db.dropCollection('test_collection_methods2');
+      expect(drop).to.be.true;
     });
 
-    it('should correctly access collection names', function (done) {
+    it('should correctly access collection names', async function () {
       // Create two collections
-      db.createCollection('test.spiderman', () => {
-        db.createCollection('test.mario', () => {
-          // Insert test documents (creates collections)
-          const spiderman_collection = db.collection('test.spiderman');
-          spiderman_collection.insertOne({ foo: 5 }, configuration.writeConcernMax(), err => {
-            expect(err).to.not.exist;
-            const mario_collection = db.collection('test.mario');
-            mario_collection.insertOne({ bar: 0 }, configuration.writeConcernMax(), err => {
-              expect(err).to.not.exist;
-              // Assert collections
-              db.collections((err, collections) => {
-                expect(err).to.not.exist;
-
-                let found_spiderman = false;
-                let found_mario = false;
-                let found_does_not_exist = false;
-
-                collections.forEach(collection => {
-                  if (collection.collectionName === 'test.spiderman') {
-                    found_spiderman = true;
-                  }
-                  if (collection.collectionName === 'test.mario') found_mario = true;
-                  if (collection.collectionName === 'does_not_exist') found_does_not_exist = true;
-                });
-
-                expect(found_spiderman).to.be.true;
-                expect(found_mario).to.be.true;
-                expect(found_does_not_exist).to.be.false;
-                done();
-              });
-            });
-          });
-        });
-      });
+      const spiderman_collection = await db.createCollection('test.spiderman');
+      const mario_collection = await db.createCollection('test.mario');
+      // Insert test documents (creates collections)
+      await spiderman_collection.insertOne({ foo: 5 }, configuration.writeConcernMax());
+      await mario_collection.insertOne({ bar: 0 }, configuration.writeConcernMax());
```

Comment:
It's interesting that these inserts were part of the test, the createCollection command will have created the collections, we don't need the inserts to do the implicit creation server side. so we can remove these.